network/wsPeer.go

```diff
@@ -534,11 +545,40 @@ func (wp *wsPeer) handleFilterMessage(msg IncomingMessage) {
 	}
 	var digest crypto.Digest
 	copy(digest[:], msg.Data)
-	//wp.net.log.Debugf("add filter %v", digest)
+	wp.net.log.Debugf("add filter %v", digest)
 	wp.outgoingMsgFilter.CheckDigest(digest, true, true)
 }
 
-func (wp *wsPeer) writeLoopSend(msg sendMessage) disconnectReason {
+var emptyHash = crypto.Digest{}
+
+func (wp *wsPeer) writeLoopSend(msgs sendMessages) disconnectReason {
+
+	// wait for another peer to finish sending before sending
+	if msgs.pacer != nil {
+		<-msgs.pacer
+
+		defer func (pacer chan int) {
+			pacer <- 1
+		}(msgs.pacer)
+	}
+
```

Comment:
I'm not a big fan of the pacer concept. It's not scalable, and slow peers would temporarily cause faster peers to get "hang". The better way to approach that is with a callback method.

Reply:
@yossigi maybe it makes sense to leave the pacer changes out then?